static/src/javascripts/projects/common/modules/commercial/krux.js

```diff
@@ -86,7 +58,7 @@ const retrieve = (n: string): string => {
 
 export const getKruxSegments = (): Array<string> => {
     const wantPersonalisedAds: boolean =
-        getAdConsentState(thirdPartyTrackingAdConsent) !== false;
+        consentState('advertisement') !== false;
```

Comment:
Why are you retreiving the `advertisment` state here directly to decide what to return, then elsewhere you are using the callback method to mutate a module level variable and you just always return the variable?

For example, above vs [here](https://github.com/guardian/frontend/pull/21696/files#diff-bcf7ccd6113bf6ff177c6ae998459b58R253).

Reply:
@ripecosta the only place `getKruxSegments` is called from is within `buildPageTargetting`, `buildPageTargetting` is only executed within the `onConsentNotification` callback defined in `build-page-targeting.js`, so we already pass the `adConsentState` into `buildPageTargetting` whenever the `onConsentNotification` callback is executed - `buildPageTargetting(adConsentState: boolean)`. 

Could we not pass the `adConsentState` on to `getKruxSegments` as so `getKruxSegments(adConsentState)`?

Reply:
Would avoid an extra callback and increase in complexity in krux.js while still making it clear to any future modules that want to use it that you're supposed to supply it with an updated version of consent. I like that idea a lot. Great catch. Thanks!

Reply:
PS: I'll do the same on user-ad-targeting.js